core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -509,6 +511,13 @@ private[spark] class ExecutorAllocationManager(
   private def onExecutorBusy(executorId: String): Unit = synchronized {
     logDebug(s"Clearing idle timer for $executorId because it is now running a task")
     removeTimes.remove(executorId)
+
+    // Executor is added to remove by misjudgment due to async listener making it as idle).
+    // see SPARK-9552
+    if (executorsPendingToRemove.contains(executorId)) {
```

Comment:
What I meant is that since this class doesn't call `killExecutors` with multiple IDs, your question about updating `executorsPendingToRemove` in that case does not apply.

`killExecutors` should return true if at least one executor was killed; it's not optimal but it fits with the current API. But that doesn't affect `executorsPendingToRemove` at all.